cbmc-ci/switch-private-submodules

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+import argparse
+import logging
+import os
+import subprocess
+import sys
+
+
+DESCRIPTION = "Switch between public and private versions of submodules"
+
+MODULES = [{
+    "submodule": "aws-encryption-sdk-cpp/tests/test_vectors/aws-encryption-sdk-test-vectors",
+    "private": "https://github.com/awslabs/private-aws-encryption-sdk-test-vectors-staging.git",
+    "public": "https://github.com/awslabs/aws-encryption-sdk-test-vectors.git",
+}, {
+    "submodule": "aws-encryption-sdk-specification",
+    "private": "https://github.com/awslabs/private-aws-encryption-sdk-specification-staging.git",
+    "public": "https://github.com/awslabs/aws-encryption-sdk-specification.git",
+}]
+
+
+def switch_to(version):
+    logging.info("Switching to %s version of the submodules", version)
+    for module in MODULES:
+        cmd = [
+            "git", "config", "--file=.gitmodules",
```

Comment:
So I see that this modifies the `.gitmodule` file itself locally. That's not ideal if we're asking developers to run this themselves locally, since then they have a spurious edit they have to ignore.

How about this instead, since it only modifies `.git/config`?

```
git config url."<private url>".insteadOf <public url>
```

https://git-scm.com/docs/git-config#Documentation/git-config.txt-urlltbasegtinsteadOf